src/libtego_ui/ui/qml/MainWindow.qml

```diff
@@ -119,6 +119,14 @@ ApplicationWindow {
             property QtObject currentContact: (visible && width > 0) ? contactList.selectedContact : null
             onCurrentContactChanged: {
                 if (currentContact !== null) {
+
+                    // remove chat page for user when they are deleted
+                    if(typeof currentContact.contactDeletedCallbackAdded === 'undefined') {
+                        currentContact.contactDeleted.connect(function(user) {
+                            remove(user.contactID);
+                        });
+                        currentContact.contactDeletedCallbackAdded = true;
```

Comment:
is this needed since the contact is removed on line 126?

Reply:
This code adds a callback to remove the contact when the contactDeleted signal is fired. The 'remove' calls is the counterpart to 'add' (which is called from within 'show' on line 130. The check for and definition of contactDeletedCallbackAdded is to ensure we only call remove once on deletion, rather than N times.